Dockerfile

```diff
@@ -63,3 +68,21 @@ ENTRYPOINT ["/docker-entrypoint.sh"]
 EXPOSE 3000
 
 CMD ["bundle", "exec", "puma"]
+
+# ------------------------------------------------------------------------------
+# test stage
+# ------------------------------------------------------------------------------
+
+FROM release as test
+
+RUN apt-get install -yy firefox-esr
+
+ARG gecko_driver_version=0.26.0
+
+RUN wget https://github.com/mozilla/geckodriver/releases/download/v$gecko_driver_version/geckodriver-v$gecko_driver_version-linux64.tar.gz
```

Comment:
On your question around security: I think this is okay. The package is downloaded from a reputable place and over HTTPS. Should it be compromised in any way in the future the impact would be limited to test runs of the image given the use of docker stages. Code installed here cannot make it into the same image used to run production code.

Reply:
Thanks for your view!